turbine/src/cluster_nodes.rs

```diff
@@ -328,6 +334,30 @@ fn get_nodes(cluster_info: &ClusterInfo, stakes: &HashMap<Pubkey, u64>) -> Vec<N
     // Since sorted_by_key is stable, in case of duplicates, this
     // will keep nodes with contact-info.
     .dedup_by(|a, b| a.pubkey() == b.pubkey())
+    .filter_map(|node| {
+        if !should_dedup_addrs
+            || node
+                .contact_info()
+                .and_then(|node| node.tvu(Protocol::UDP).ok())
+                .map(|addr| {
+                    *counts
+                        .entry(addr.ip())
+                        .and_modify(|count| *count += 1)
+                        .or_insert(1)
+                })
```

Comment:
I get lifetime errors with an  out-of-line closure.

Reply:
:thinking: 

are you capturing `counts` instead of passing it in?

Reply:
Passing it might work but it definitely becomes a more verbose code.
I think the inline version is better anyways because it is explicit what is it doing and don't need to jump elsewhere to figure that out.